### PR TITLE
[DEV-1018] add yield machine deposit

### PR DESCRIPTION
**Pull request by Krysh90**

### CODE REVIEW

File: src/subdomains/payin/infrastructure/payin-crypto-defichain.service.ts [hunk 1]
```diff
@@ -13,11 +13,14 @@ interface HistoryAmount {
...
 }
 
+type AccountHistory = Omit<JellyAccountHistory & HistoryAmount, 'amounts'>;
+
 @Injectable()
 export class PayInDeFiChainService {
   private client: DeFiClient;
 
   private readonly utxoTxTypes = ['receive', 'AccountToUtxos'];
+  private readonly tokenTxTypes = ['AccountToAccount'];
```
Comment (davidleomay) [inline src/subdomains/payin/infrastructure/payin-crypto-defichain.service.ts:23]: Missing token types

File: src/subdomains/staking/application/services/staking.service.ts [hunk 2]
```diff
@@ -52,11 +54,18 @@ export class StakingService {
 
     const { asset: assetName, blockchain } = query;
 
+    // fallback handling for non adapted apps
+    const strategy = query.strategy ?? StakingStrategy.MASTERNODE;
+    if (!query.strategy) query.strategy = StakingStrategy.MASTERNODE;
```
Comment (davidleomay) [inline src/subdomains/staking/application/services/staking.service.ts:59]: Do this in the DTO directly? We could then use `@IsNotEmpty()`.
```ts
  @ApiProperty()
  @IsNotEmpty()
  @IsEnum(StakingStrategy)
  strategy: StakingStrategy = StakingStrategy.MASTERNODE;
```

File: src/subdomains/staking/application/services/staking.service.ts [hunk 3]
```diff
@@ -169,10 +178,10 @@ export class StakingService {
     const withdrawalAddress = await this.userService.getWalletAddress(userId, walletId);
 
     // only one staking per address
-    const existingStaking = await this.repository.findOne({ where: { withdrawalAddress } });
+    const existingStaking = await this.repository.findOne({ where: { withdrawalAddress, strategy: dto.strategy } });
```
Comment (davidleomay) [inline src/subdomains/staking/application/services/staking.service.ts:181]: Also add the asset here? Is not really used right now, but would be "more correct".

File: src/subdomains/staking/application/validators/staking-strategy.validator.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { Injectable } from '@nestjs/common';
+import { Asset } from 'src/shared/models/asset/asset.entity';
+import { StakingStrategy } from '../../domain/enums';
+
+@Injectable()
```
Comment (davidleomay) [inline src/subdomains/staking/application/validators/staking-strategy.validator.ts:5]: Use a class with static method instead of injectable service?

**Review by davidleomay — COMMENTED** [review]
image

**Comment by Krysh90** [comment]
> image

removed

**Review by davidleomay — APPROVED** [review]

**Review by davidleomay — COMMENTED** [review]
Better wait with this PR until issue [DEV-1040](https://dfxswiss.atlassian.net/browse/DEV-1040) is done.

**Pull request merged**